src/validation/rules/OverlappingFieldsCanBeMergedRule.ts

```diff
@@ -469,6 +483,44 @@ function findConflictsBetweenSubSelectionSets(
   return conflicts;
 }
 
+// (E) Then collect any conflicts between the provided collection of fields
+// and any fragment names found in the given fragment.
+const processDiscoveredFragments = (
+  context: ValidationContext,
+  conflicts: Array<Conflict>,
+  cachedFieldsAndFragmentNames: Map<SelectionSetNode, FieldsAndFragmentNames>,
+  comparedFragmentPairs: PairSet,
+  areMutuallyExclusive: boolean,
+  fieldMap: NodeAndDefCollection,
+  discoveredFragments: Array<Array<string>>,
+) => {
+  while (discoveredFragments.length !== 0) {
```

Comment:
tiny suggestion to make TS happy without additional check

```suggestion
  let item;
  while ((item = discoveredFragments.pop()) !== undefined) {
    if (comparedFragmentPairs.has(item[1], item[0], areMutuallyExclusive)) {
```